### PR TITLE
 make openFirewall options to false for NixOS services

### PR DESCRIPTION
**Pull request by rapenne-s**
###### Description of changes

openFirewall should be set to false all the time, except for openssh, this avoids surprises to users who may not want to expose an enabled service without explicit consent.

Pardon me for my rebase mistake in the previous PR involving 450 commits :pray: 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 1]
```diff
@@ -27,6 +27,14 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - `services.sourcehut.dispatch` and the corresponding package (`sourcehut.dispatchsrht`) have been removed due to [upstream deprecation](https://sourcehut.org/blog/2022-08-01-dispatch-deprecation-plans/).
 
+- The `snapserver` module option `openFirewall` default value has been changed from `true` to `false`. You will need to explicitely set this option to `true`, or configure your firewall.
```
Comment (Ma27) [inline nixos/doc/manual/release-notes/rl-2305.section.md:30]: Nit: I'm a fan of linking options whenever referencing them. I.e. to write `[services.snapserver.openFirewall](#opt-services.snapserver.openFirewall)`. With that it's possible for the reader to directly navigate to the option's manual.

File: nixos/modules/services/audio/snapserver.nix [hunk 2]
```diff
@@ -101,9 +101,7 @@ in {
 
       openFirewall = mkOption {
         type = types.bool;
-        # Make the behavior consistent with other services. Set the default to
-        # false and remove the accompanying warning after NixOS 22.05 is released.
-        default = true;
+        default = false;
```
Comment (Ma27) [inline nixos/modules/services/audio/snapserver.nix:104]: I guess the warning below can be removed here as well.

**Review by Ma27 — COMMENTED** [review]
Two minor notes, otherwise LGTM, thanks!

File: nixos/doc/manual/release-notes/rl-2305.section.md (hunk 1 above)
Comment (rapenne-s) [reply nixos/doc/manual/release-notes/rl-2305.section.md:30]: I'll do that, and this even caught an issue! The module for avahi is `avahi` and not `avahi-daemon`.

**Review by tobim — APPROVED** [review]

**Comment by tobim** [comment]
Thanks!

**Review by Ma27 — APPROVED** [review]

**Pull request merged**